Source/WebCore/platform/graphics/cg/PathCG.cpp

```diff
@@ -21,167 +21,316 @@
+    case kCGPathElementMoveToPoint:
+        applier({ PathMoveTo { cgPoints[0] } });
+        break;
+
+    case kCGPathElementAddLineToPoint:
+        applier({ PathLineTo { cgPoints[0] } });
+        break;
+
+    case kCGPathElementAddQuadCurveToPoint:
+        applier({ PathQuadCurveTo { cgPoints[0], cgPoints[1] } });
+        break;
+
+    case kCGPathElementAddCurveToPoint:
+        applier({ PathBezierCurveTo { cgPoints[0], cgPoints[1], cgPoints[2] } });
+        break;
+
+    case kCGPathElementCloseSubpath:
+        applier({ std::monostate() });
+        break;
     }
-    m_inlineData = std::monostate { };
-    return m_path.get();
 }
 
-bool Path::isNull() const
+void PathCG::applySegments(const PathSegmentApplier& applier) const
 {
-    return !m_path && !hasInlineData();
+    CGPathApply(platformPath(), (void*)&applier, pathSegmentApplierCallback);
 }
 
-Path::Path(const Path& other)
+static void pathElementApplierCallback(void* info, const CGPathElement* element)
 {
-    m_path = { other.m_path };
-    m_inlineData = other.m_inlineData;
-    if (m_path) {
-        m_copyPathBeforeMutation = true;
-        other.m_copyPathBeforeMutation = true;
+    const auto& applier = *(PathElementApplier*)info;
+    auto* cgPoints = element->points;
+
+    switch (element->type) {
+    case kCGPathElementMoveToPoint:
+        applier({ PathElement::Type::MoveToPoint, { cgPoints[0] } });
+        break;
+
+    case kCGPathElementAddLineToPoint:
+        applier({ PathElement::Type::AddLineToPoint, { cgPoints[0] } });
+        break;
+
+    case kCGPathElementAddQuadCurveToPoint:
+        applier({ PathElement::Type::AddQuadCurveToPoint, { cgPoints[0], cgPoints[1] } });
+        break;
+
+    case kCGPathElementAddCurveToPoint:
+        applier({ PathElement::Type::AddCurveToPoint, { cgPoints[0], cgPoints[1], cgPoints[2] } });
+        break;
+
+    case kCGPathElementCloseSubpath:
+        applier({ PathElement::Type::CloseSubpath, { } });
+        break;
     }
 }
 
-Path::Path(Path&& other)
-    : m_path(std::exchange(other.m_path, nullptr))
-    , m_inlineData(std::exchange(other.m_inlineData, std::monostate { }))
-    , m_copyPathBeforeMutation(std::exchange(other.m_copyPathBeforeMutation, false))
+void PathCG::applyElements(const PathElementApplier& applier) const
 {
+    CGPathApply(platformPath(), (void*)&applier, pathElementApplierCallback);
 }
 
-void Path::swap(Path& otherPath)
+bool PathCG::isEmpty() const
 {
-    std::swap(m_path, otherPath.m_path);
-    std::swap(m_inlineData, otherPath.m_inlineData);
-    std::swap(m_copyPathBeforeMutation, otherPath.m_copyPathBeforeMutation);
+    return CGPathIsEmpty(platformPath());
 }
 
-Path& Path::operator=(const Path& other)
+size_t PathCG::elementCount() const
 {
-    Path copy { other };
-    swap(copy);
-    return *this;
+#if HAVE(CGPATH_GET_NUMBER_OF_ELEMENTS)
+    return CGPathGetNumberOfElements(platformPath());
+#else
+    return PathImpl::elementCount();
+#endif
 }
```

Comment:
If we do end up removing `Path::elementCount`, then we can also remove `CGPathGetNumberOfElements` from CoreGraphicsSPI.h.

Reply:
`CGPathGetNumberOfElements` was removed from CoreGraphicsSPI.h